frontend/src/components/pages/CamperRefundCancellation/index.tsx

```diff
@@ -25,27 +28,58 @@ const CamperRefundCancellation = (): React.ReactElement => {
   const toast = useToast();
   const [refunds, setRefunds] = useState<RefundDTO>([]);
   const [campName, setCampName] = useState<string>("");
+  const [validCode, setValidCode] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(true);
 
   // The camper-refund-cancellation route will have an id to identify the refund code
   const { id: refundCode } = useParams<{ id: string }>();
 
   useEffect(() => {
     const getRefundInfoById = async (code: string) => {
-      const getResponse = await CamperAPIClient.getRefundInfo(code);
-      if (getResponse) {
+      try {
+        const getResponse = await CamperAPIClient.getRefundInfo(code);
+        setValidCode(true);
         setRefunds(getResponse);
         setCampName(getResponse[0].campName);
-      } else {
+      } catch {
+        setValidCode(false);
```

Comment:
Might not be necessary if it is initialized at false and never goes from true to false